custom/icds/location_reassignment/utils.py

```diff
@@ -90,3 +91,26 @@ def get_child_case_ids(ids):
         else:
             case_ids = None
     return all_child_case_ids
+
+
+@task(queue=settings.CELERY_LOCATION_REASSIGNMENT_QUEUE)
+def update_usercase(domain, old_username, new_username):
+    if "@" not in old_username:
+        old_username = normalize_username(old_username, domain)
+    if "@" not in new_username:
+        new_username = normalize_username(new_username, domain)
+    old_user = CouchUser.get_by_username(old_username)
+    new_user = CouchUser.get_by_username(new_username)
+    if old_user and new_user and old_user.is_commcare_user() and new_user.is_commcare_user():
+        old_user_usercase = old_user.get_usercase()
+        new_user_usercase = new_user.get_usercase()
+        # pick values that are not already present on the new user's usercase, populated already via HQ
```

Comment:
Why doesn't this also update keys that are present on both the old and new usercases but have different values?

Reply:
I am not fully clear on what all updates should be done but starting with the ones that are missing. For the ones already present, my understanding is that the ones already present are set according to the new user and should not be changed?

Reply:
Ah, I get it now, yeah that sounds right.